### ISSUE TITLE
False positive: unreachable

### ISSUE THREAD

**Issue by KotlinIsland** [python/mypy#10806]
**Bug Report**

false positive unreachable

**To Reproduce**

```py
a: object = None

def foo() -> None:
    global a
    a = None
    bar()
    if a:
        print("hi")

def bar() -> None:
    global a
    a = 1
```
**Expected Behavior**

no error

**Actual Behavior**

main.py:8: error: Statement is unreachable
Found 1 error in 1 file (checked 1 source file)

(Write what happened.)

**Your Environment**
https://mypy-play.net/?mypy=latest&python=3.10&flags=warn-unreachable&gist=c41cd432e95c591743aa1369f36823d1

**Comment by Akuli** [comment]
related: #10634
